src/pages/api/entry/[user_id]/[date].ts

```diff
@@ -0,0 +1,34 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { getSupabase } from '../../../../utils/supabase'
+import { PostgrestError } from '@supabase/supabase-js';
+
+export type SingleEntryData = {
+  data: { flow: string | null, mood: string | null, craving: string | null, symptom: string | null, user_id: string | null, date: string} | null
+}
+
+export default async function getEntry(
+  req: NextApiRequest,
+  res: NextApiResponse<SingleEntryData | PostgrestError | Error>
+) {
+  const { date, user_id } = req.query;
```

Comment:
we already spoke about this but this weekend I'd like to take on refactoring so that we don't need to pass `userID` from the client to the BE